File: capacity-club-api/src/domain-modules/address/model/payload/address-create.payload.ts
import { ApiCodeResponse } from '@common/api';
import { addressProperties } from './../../../../common/documentation/swagger.annotations';
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString, IsOptional } from 'class-validator';

/**
 * Class used to create address.
 * This class defines the payload used when creating new address.
 */
export class AddressCreatePayload {
  @ApiProperty({
    name: addressProperties.street.name,
    description: addressProperties.street.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.ADDRESS_PAYLOAD_STREET_IS_MISSING,
  })
  @IsString()
  street: string;
  @ApiProperty({
    name: addressProperties.number.name,
    description: addressProperties.number.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.ADDRESS_PAYLOAD_NUMBER_IS_MISSING,
  })
  @IsString()
  number: string;
  @ApiProperty({
    name: addressProperties.zipcode.name,
    description: addressProperties.zipcode.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.ADDRESS_PAYLOAD_ZIPCODE_IS_MISSING,
  })
  @IsString()
  zipcode: string;
  @ApiProperty({
    name: addressProperties.town.name,
    description: addressProperties.town.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.ADDRESS_PAYLOAD_TOWN_IS_MISSING,
  })
  @IsString()
  town: string;
  @ApiProperty({
    name: addressProperties.country.name,
    description: addressProperties.country.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.ADDRESS_PAYLOAD_COUNTRY_IS_MISSING,
  })
  @IsString()
  country: string;
  @ApiProperty({
    name: addressProperties.complement.name,
    description: addressProperties.complement.description,
  })
  @IsOptional()
  @IsString()
  complement: string;
}

File: capacity_club_mobile_app/lib/domain-features/data/address/model/payload/address_create_payload.dart
import 'package:capacity_club_mobile_app/core/model/abstract/payload.dart';
import 'package:equatable/equatable.dart';

class AddressCreatePayload extends Equatable implements Payload {
  final String street;
  final String number;
  final String zipcode;
  final String town;
  final String country;
  final String complement;

  AddressCreatePayload({
    required this.street,
    required this.number,
    required this.zipcode,
    required this.town,
    required this.country,
    this.complement = '',
  });

  Map<String, dynamic> toJson() => {
        'street': street,
        'number': number,
        'zipcode': zipcode,
        'town': town,
        'country': country,
        'complement': complement,
      };

  @override
  List<Object?> get props =>
      [street, number, zipcode, town, country, complement];
}
